test/src/main/java/io/strimzi/test/TestUtils.java

```diff
@@ -120,9 +121,11 @@ public static long waitFor(String description, long pollIntervalMs, long timeout
             }
             if (timeLeft <= 0) {
                 onTimeout.run();
-                TimeoutException exception = new TimeoutException("Timeout after " + timeoutMs + " ms waiting for " + description);
-                exception.printStackTrace();
-                throw exception;
+                try {
+                    throw new TimeoutException("Timeout after " + timeoutMs + " ms waiting for " + description);
```

Comment:
Are you throwing new exception on line 125 to catch it on 126? Can you explain the point of it?

Reply:
I have to change it, that was not a correct way. Anyway, the idea of `io.strinzi.test.TimeoutException` in this case was to replace the original TimeoutException from the concurrent java package.  I have replaced it to the RuntimeException which is from my POV a proper way.  

The best way would be using the TimeoutException from the concurrent package but I can not use it in that way because it needs to be inherited from the RuntimeException.

Is there something I missing @tombentley  ?

Reply:
IIRC I created `io.strinzi.test.TimeoutException` because I needed/wanted an unchecked exception to represent a timeout.

Reply:
and it is worth to have it and not just using the `RuntimeException`, which satisfy this condition of `unchecked exception`?

Reply:
Well it depends how useful you think it is to see (and potentially `catch`) a `TimeoutException` rather than a `RuntimeException`. To me a `TimeoutException` is clearer because you know, even before looking at the log message, that some expectation was not met. `RuntimeException` is much more general and requires you to interpret the message to figure out what went wrong. Why do you think `RuntimeException` is better?

Reply:
My point here is to remove the dependency of our io.strimzi.test.TimeException, because in our tests, we sometimes met with the problem conflict class. For instance, most of the standard classes use `concurrent.TimeoutException`. I accept that TimeoutException inherited from the RuntimeException are making more sense in terms of semantic. However, I think we always need to make sure which exception is imported in some of our `STs`. Hypothetically, we also can meet the scenario wherein the one `STs` will have both exceptions and this is from my POV, not a good approach.  Is that making any sense? @tombentley

Reply:
If it's a nuisance having two classes called `TimeoutException` you could just rename the Strimzi one to `WaitException`.

Reply:
Done :)